rest/config.go

```diff
@@ -1399,6 +1400,11 @@ func (sc *ServerContext) fetchDatabase(ctx context.Context, dbName string) (foun
 			continue
 		}
 		if err != nil {
+			if err.(*json.UnmarshalTypeError) != nil {
+				base.WarnfCtx(ctx, "unable to fetch config in group %q from bucket %q: %v", sc.Config.Bootstrap.ConfigGroupID, bucket, err)
+				cnf.Bucket = &bucket
+				return true, &cnf, nil
+			}
```

Comment:
This seems overly specific to one particular type of broken config error, and only handling one json library in particular. EE uses a non-stdlib json package, which returns different error types.

I'm thinking we'll have to refactor to split this function into two, so you're able to use the pieces you need from it, without having to unmarshal into a full DatabaseConfig struct.

My initial thoughts are to write a function that does the existing loop logic and invoke a callback.
The callback has the ability to either break the loop or continue and is given `bucket` as a parameter, so it's still able to call stuff like `GetConfig(bucket, ...)`

Then two callbacks:
- One that uses the existing inner-loop code which is used by fetchDatabase
- Another specific to your use case: finding the bucket name that matches a given db name, if one exists...

For this we don't need to unmarshal into a full DatabaseConfig. We can unmarshal into an inline struct to only get what we need (and ignore any of the other invalid properties we may have:

```go
func (sc *ServerContext) foreachDbConfig(ctx, callback func(bucket string) (exit bool)) {
	// existing for each bucket/looping code and callback
}

// functionally same as before
func (sc *ServerContext) fetchDatabase(ctx, dbName) {
	// sc.foreachDbConfig ...
	// callback implementing existing logic
}


func (sc *ServerContext) bucketNameFromDbName(dbName string) (bucketName string, found bool) {
	// sc.foreachDbConfig ...
	// callback implementing your new logic
		var cfgDbName struct {
				Name string `json:"name"`
		}
		cas, err := sc.BootstrapContext.Connection.GetConfig(bucket, sc.Config.Bootstrap.ConfigGroupID, &cfgDbName)
		...
		if dbName == cfgDbName.Name {
			return bucket
		}
}
```

Reply:
[d13063f](https://github.com/couchbase/sync_gateway/pull/5894/commits/d13063f8f0e9cb90de6f167b6ead9199b55c4d5b) addresses this comment, let me know if further changes are required